lib/tasks/grade.rake

```diff
@@ -203,20 +236,29 @@ def upstream_repo(root_url, token)
   res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
     http.request(req)
   end
-  result = Oj.load(res.body)
-  result["repo_slug"]
+  Oj.load(res.body)
 rescue => e
   return false
 end
 
-def github_username(primary_email)
-  return "" if primary_email.blank?
-  username = `git config user.name`.chomp
-  search_results = Octokit.search_users("#{primary_email} in:email").fetch(:items)
-  if search_results.present?
-    username = search_results.first.fetch(:login, username)
+def retrieve_github_username
+  config_dir_name = find_or_create_directory(".vscode")
+  config_file_name = "#{config_dir_name}/.ltici_apitoken.yml"
+  if File.exist?(config_file_name)
+    config = YAML.load_file(config_file_name)
+    if config["github_username"].present?
+      return config["github_username"]
+    end
```

Comment:
Is it ever the case that the username is written to the `.ltici_apitoken.yml` file? I don't see where in the logic this would be happening, and in my testing I don't see it being written there. Just wondering if / why this is necessary.

Reply:
Good catch. I think I accidentally removed this from regularly running `rake grade`. It does get set when running `rake grade:reset_token` but I'll make sure it gets set on the first `grade` task.